interpreter.py

```diff
@@ -7,12 +7,18 @@ def __init__(self, logger):
         self.direction = [1, 0]
         self.holding = []
 
-    def evaluate_block(self, commands):
-        for cmd in commands:
-            cmd.accept(self)
+    def evaluate_block(self, count, commands):
+        #using this count, interate through each command in "commands" list
+        #ie: 3[FR] calls the sequence "FR" three times
+        for cmdcountiter in range(0, int(count)):
+            for cmd in commands:
+                cmd.accept(self)
```

Comment:
I don't see why block has to have its own counter. If a block is a kind of command, then couldn't the command execution take care of this automatically, so 3[FL] would see the command calling accept on the block 3 times.

# -2